src/nnbench/types.py

```diff
@@ -98,4 +99,22 @@ def __post_init__(self):
             name = self.fn.__name__
 
             super().__setattr__("name", name)
-        # TODO: Parse interface using `inspect`, attach to the class
+            super().__setattr__("interface", Interface.from_callable(self.fn))
+
+
+@dataclass(frozen=True)
+class Interface:
+    varnames: tuple[str, ...]
+    vartypes: tuple[type, ...]
+    varitems: tuple[tuple[str, inspect.Parameter], ...]
```

Comment:
```suggestion
    varitems: tuple[tuple[str, type], ...]
```

(Maybe you can even do a type alias `Variable = tuple[str, type]` and then go `variables: tuple[Variable, ...]` with the suggestion below.)